docs/setup_installation/azure/getting_started.md

```diff
@@ -24,78 +24,137 @@ To run all the commands on this page the user needs to have at least the followi
+Create a custom role definition with the minimum permissions needed to read and write to the storage account:
 
 ```bash
-export managed_id=`az aks show --resource-group $resource_group --name $cluster_name --query "identity.principalId" --output tsv`
+export STORAGE_ID=`az storage account show --name $STORAGE_ACCOUNT_NAME --resource-group $RESOURCE_GROUP --query "id" --output tsv`
+
+az role definition create --role-definition '{
+  "Name": "hopsfs-storage-permissions",
+  "IsCustom": true,
+  "Description": "Allow HopsFS to access the storage container",
+  "Actions": [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/write",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/read",
+    "Microsoft.Storage/storageAccounts/blobServices/write",
+    "Microsoft.Storage/storageAccounts/blobServices/read",
+    "Microsoft.Storage/storageAccounts/listKeys/action"
+  ],
+  "NotActions": [],
+  "DataActions": [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/move/action",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write"
+  ],
+  "AssignableScopes": [
+    "'$STORAGE_ID'"
+  ]
+}'
+
+sleep 30  # give Azure some time to persist the new role
+
+az role assignment create --role hopsfs-storage-permissions --assignee $UA_IDENTITY_PRINCIPAL_ID --scope $STORAGE_ID
+```
 
-export storage_id=`az storage account show --name $storage_account_name --resource-group $resource_group --query "id" --output tsv`
+Also, assign roles to pull/push images from/to the container registry:
 
-export acr_id=`az acr show --name $registry_name --resource-group $resource_group --query "id" --output tsv`
+```bash
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role AcrPull --scope $ACR_ID
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role "AcrPush" --scope $ACR_ID
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role "AcrDelete" --scope $ACR_ID
 ```
 
-### Step 1.5: Assign Roles to Managed Identity
+### Step 1.5: Create Service Principal for Hopsworks services
 
-```bash
-az role assignment create --assignee $managed_id --role "Storage Blob Data Contributor" --scope $storage_id
+Create a service principal to grant Hopsworks applications with access to the container registry. For example, Hopsworks uses this service principal to push new Python environments created via the Hopsworks UI.
 
-az role assignment create --assignee $managed_id --role AcrPull --scope $acr_id
-az role assignment create --assignee $managed_id --role "AcrPush" --scope $acr_id
-az role assignment create --assignee $managed_id --role "AcrDelete" --scope $acr_id
+```bash
+export SP_PASSWORD=`az ad sp create-for-rbac --name $SP_NAME --scopes $ACR_ID --role acrpush --query "password" --output tsv`
```

Comment:
do they have a parameter to set the expiry date explicitly?

Reply:
Yes, there is a `--years` parameter that defaults to 1.